src/app/auth/auth.component.ts

```diff
@@ -0,0 +1,41 @@
+import { Component, OnInit } from '@angular/core';
+import { AuthService } from '../core/services/auth.service';
+import {Router} from '@angular/router';
+
+@Component({
+  selector: 'qpac-auth',
+  templateUrl: './auth.component.html',
+  styleUrls: ['./auth.component.scss']
+})
+export class AuthComponent implements OnInit {
+
+  constructor(
+    private auth: AuthService,
+    private router: Router
+  ) { }
+
+  ngOnInit() {
+  }
+
+  public doFbLogin() {
+    this.auth.doFbLogin().then(res => {
+      console.log(res);
+      this.router.navigate(['/dashboard']);
+    });
+  }
+
+  public doGoogleLogin() {
+    this.auth.doGoogleLogin().then(res => {
+      console.log(res);
+      this.router.navigate(['/dashboard']);
```

Comment:
According to software development principle DRY (https://en.wikipedia.org/wiki/Don%27t_repeat_yourself), we can move `this.router.navigate(['/dashboard']);` to a separate method, e.g. `navigateToDashboard()` and reuse it.